File: code/perfect-java-code/perfect-design-pattern/src/creation/factory/simplefactory/pizzastore/pizza/CheesePizza.java
package creation.factory.simplefactory.pizzastore.pizza;

/**
 * @author Kelly
 * @create 2020-09-18 16:32
 */
public class CheesePizza extends Pizza  {
    @Override
    public void prepare() {
        System.out.println("给【奶酪披萨】准备原材料...");
    }
}

File: code/perfect-java-code/perfect-scala/src/main/scala/com/learn/chapter17/simplefactory/pizzastore/pizza/CheesePizza.scala
package com.learn.chapter17.simplefactory.pizzastore.pizza

/**
 * @author Kelly
 * @create 2021-05-13 15:36
 */
class CheesePizza extends Pizza {
  override def prepare(): Unit = {
    this.name = "Cheese Pizza"
    println(this.name + " preparing")
  }
}
